plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -191,6 +205,72 @@ public TableSnapshot loadSnapshot(ConnectorSession session, SchemaTableName tabl
         return snapshot;
     }
 
+    private TableSnapshot loadSnapshotForTimeTravel(TrinoFileSystem fileSystem, SchemaTableName table, String tableLocation, long endVersion)
```

Comment:
We should actually get the endVersion first, then the previous version , ..., until we find the checkpoint .

I'm thinking about something similar to `io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail#loadNewTail` but backwards.

Doing this would get us to retrieve exactly what we need and not do any unnecessary file listings in the transaction log directory.

Reply:
The suggested logic works for classic checkpoint, but it might be difficult in case of multipart checkpoints and UUID-named checkpoints. We can't estimate the file name in my understanding. What do you think?

Reply:
Probably, we can use different logic based on `delta.checkpointPolicy` table property in the future. 
https://github.com/delta-io/delta/blob/537ed8ee0be983579873851855ba3e96b20004bd/docs/source/table-properties.md?plain=1#L159-L170